### PR TITLE
Add name_len field to config structures taking cache/core name

### PR DESCRIPTION
**Pull request by Ostrokrzew**

### CODE REVIEW

File: src/mngt/ocf_mngt_core.c [hunk 1]
```diff
@@ -541,8 +541,8 @@ void ocf_mngt_cache_add_core(ocf_cache_t cache,
 		goto err_pipeline;
 	}
 
-	result = env_strncpy(name, OCF_CORE_NAME_SIZE,
-			cfg->name, OCF_CORE_NAME_SIZE);
+	result = env_strncpy(name, cfg->name_len,
```
Comment (arutk) [inline src/mngt/ocf_mngt_core.c]: second param should be OCF_CORE_NAME_SIZE

File: inc/ocf_mngt.h [hunk 2]
```diff
@@ -237,6 +243,11 @@ struct ocf_mngt_cache_config {
 	 */
 	char name[OCF_CACHE_NAME_SIZE];
 
+	/**
+	 * @brief Cache name length
```
Comment (arutk) [inline inc/ocf_mngt.h]: Maybe let's use the comment from strncpy_s:
@brief The maximum number of characters to copy from name

File: inc/ocf_mngt.h [hunk 3]
```diff
@@ -60,6 +65,7 @@ struct ocf_mngt_core_config {
 static inline void ocf_mngt_core_config_set_default(
 		struct ocf_mngt_core_config *cfg)
 {
+	cfg->name_len = sizeof(*cfg->name);
```
Comment (arutk) [inline inc/ocf_mngt.h]: This is wrong. name_len is supposed to store size of the string. Please remove this from set_default function - whoever sets cache name is responsible for setting this value as well.

File: inc/ocf_mngt.h [hunk 4]
```diff
@@ -302,6 +313,7 @@ struct ocf_mngt_cache_config {
 static inline void ocf_mngt_cache_config_set_default(
 		struct ocf_mngt_cache_config *cfg)
 {
+	cfg->name_len = sizeof(*cfg->name);
```
Comment (arutk) [inline inc/ocf_mngt.h]: This is wrong. name_len is supposed to store size of the string. Please remove this from set_default function - whoever sets cache name is responsible for setting this value as well.

File: inc/ocf_mngt.h (hunk 2 above)
Comment (arutk) [reply inc/ocf_mngt.h]: same comment applies to core config

File: src/mngt/ocf_mngt_core.c (hunk 1 above)
Comment (Ostrokrzew) [reply src/mngt/ocf_mngt_core.c]: done

File: example/simple/src/main.c [hunk 5]
```diff
@@ -91,7 +91,7 @@ static void simple_complete(ocf_cache_t cache, void *priv, int error)
  */
 int initialize_cache(ocf_ctx_t ctx, ocf_cache_t *cache)
 {
-	struct ocf_mngt_cache_config cache_cfg = { .name = "cache1" };
+	struct ocf_mngt_cache_config cache_cfg = { .name = "cache1", .name_len = env_strnlen(cache_cfg.name, OCF_CACHE_NAME_SIZE) };
```
Comment (arutk) [inline example/simple/src/main.c]: name_len should be assigned "env_strnlen(...) + 1".   Or even better" sizeof("cache1")

File: example/simple/src/main.c [hunk 6]
```diff
@@ -208,6 +208,7 @@ int initialize_core(ocf_cache_t cache, ocf_core_t *core)
 	/* Core configuration */
 	ocf_mngt_core_config_set_default(&core_cfg);
 	core_cfg.name = "core1";
+	core_cfg.name_len = env_strnlen(core_cfg.name, OCF_CORE_NAME_SIZE);
```
Comment (arutk) [inline example/simple/src/main.c]: ... same here - name_len should be set to the size of string memory buffer, not the actual string length. Let's discuss live

File: src/mngt/ocf_mngt_core.c (hunk 1 above)
Comment (arutk) [reply src/mngt/ocf_mngt_core.c]: I meant second out of four, not the second one that you change :)

env_strncpy(name, OCF_CORE_NAME_SIZE, cfg->name, cfg->name_len)

File: example/simple/src/main.c [hunk 7]
```diff
@@ -91,7 +91,8 @@ static void simple_complete(ocf_cache_t cache, void *priv, int error)
  */
 int initialize_cache(ocf_ctx_t ctx, ocf_cache_t *cache)
 {
-	struct ocf_mngt_cache_config cache_cfg = { .name = "cache1" };
+	struct ocf_mngt_cache_config cache_cfg = { .name = "cache1",
+											   .name_len = sizeof("cache1") };
```
Comment (Ostrokrzew) [inline example/simple/src/main.c]: in editor this line starts under '.name'

File: src/mngt/ocf_mngt_core.c (hunk 1 above)
Comment (Ostrokrzew) [reply src/mngt/ocf_mngt_core.c]: my mistake, now fixed

File: example/simple/src/main.c (hunk 5 above)
Comment (Ostrokrzew) [reply example/simple/src/main.c]: changed

File: example/simple/src/main.c (hunk 6 above)
Comment (Ostrokrzew) [reply example/simple/src/main.c]: fixed and looking for discussion

File: example/simple/src/main.c (hunk 7 above)
Comment (robertbaldyga) [reply example/simple/src/main.c]: So probably you have tab length set to 4 chars instead of 8.

Comment (Ostrokrzew) [reply example/simple/src/main.c]: that was true

**Pull request closed without merging**